File: app/src/main/java/info/nightscout/androidaps/plugins/DanaR/comm/MsgSettingShippingInfo.java
package info.nightscout.androidaps.plugins.DanaR.comm;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.Date;

import info.nightscout.androidaps.Config;
import info.nightscout.androidaps.plugins.DanaR.DanaRPlugin;

/**
 * Created by mike on 05.07.2016.
 */
public class MsgSettingShippingInfo extends MessageBase {
    private static Logger log = LoggerFactory.getLogger(MsgSettingShippingInfo.class);

    public MsgSettingShippingInfo() {
        SetCommand(0x3207);
    }

    public void handleMessage(byte[] bytes) {
        DanaRPlugin.getDanaRPump().serialNumber = stringFromBuff(bytes, 0, 10);
        DanaRPlugin.getDanaRPump().shippingDate = dateFromBuff(bytes, 10);
        DanaRPlugin.getDanaRPump().shippingCountry = asciiStringFromBuff(bytes, 13, 3);
        if (DanaRPlugin.getDanaRPump().shippingDate.getTime() > new Date(116, 4, 1).getTime()) {
            DanaRPlugin.getDanaRPump().isNewPump = true;
        } else
            DanaRPlugin.getDanaRPump().isNewPump = false;
        if (Config.logDanaMessageDetail) {
            log.debug("Serial number: " + DanaRPlugin.getDanaRPump().serialNumber);
            log.debug("Shipping date: " + DanaRPlugin.getDanaRPump().shippingDate);
            log.debug("Shipping country: " + DanaRPlugin.getDanaRPump().shippingCountry);
        }
    }
}

File: pump/danar/src/main/kotlin/app/aaps/pump/danar/comm/MsgSettingShippingInfo.kt
package app.aaps.pump.danar.comm

import app.aaps.core.interfaces.logging.LTag
import app.aaps.core.interfaces.di.MetroMemberInjector

class MsgSettingShippingInfo(
    injector: MetroMemberInjector
) : MessageBase(injector) {

    init {
        setCommand(0x3207)
        aapsLogger.debug(LTag.PUMPCOMM, "New message")
    }

    override fun handleMessage(bytes: ByteArray) {
        danaPump.serialNumber = stringFromBuff(bytes, 0, 10)
        danaPump.shippingDate = dateFromBuff(bytes, 10)
        danaPump.shippingCountry = asciiStringFromBuff(bytes, 13, 3)
        aapsLogger.debug(LTag.PUMPCOMM, "Serial number: " + danaPump.serialNumber)
        aapsLogger.debug(LTag.PUMPCOMM, "Shipping date: " + danaPump.shippingDate)
        aapsLogger.debug(LTag.PUMPCOMM, "Shipping country: " + danaPump.shippingCountry)
    }
}